src/network.ts

```diff
@@ -53,23 +53,36 @@ export async function launchNetwork() {
     }
     client.on(NetworkClient.Events.TRANSACTION, transactionListener);
 
-    // Subscribe to new addresses (for balance updates and transactions)
     const subscribedAddresses = new Set<string>();
+    const fetchedAddresses = new Set<string>();
+
+    // Subscribe to new addresses (for balance updates and transactions)
+    // Also remove logged out addresses from fetched (so that they get fetched on next login)
     watch(addressStore.addressInfos, () => {
         const newAddresses: string[] = [];
+        const removedAddresses: Set<string> = new Set<string>(subscribedAddresses);
```

Comment:
Both type annotations are unnecessary here, as TS can infer that creating a new set from a `Set<string>` creates a `Set<string>`.

Reply:
Haha, I've been doing too much Java lately and unconsciously putting types everwhere!